src/Compute/Compute/Generated/DedicatedHost/DedicatedHostDeleteMethod.cs

```diff
@@ -0,0 +1,135 @@
+                    string hostName;
+                    switch (this.ParameterSetName)
+                    {
+                        case "ResourceIdParameter":
+                            resourceGroupName = GetResourceGroupName(this.ResourceId);
+                            hostGroupName = GetResourceName(this.ResourceId, "Microsoft.Compute/hostGroups", "hosts");
+                            hostName = GetInstanceId(this.ResourceId, "Microsoft.Compute/hostGroups", "hosts");
+                            break;
+                        case "ObjectParameter":
+                            resourceGroupName = GetResourceGroupName(this.InputObject.Id);
+                            hostGroupName = GetResourceName(this.InputObject.Id, "Microsoft.Compute/hostGroups", "hosts");
+                            hostName = GetInstanceId(this.InputObject.Id, "Microsoft.Compute/hostGroups", "hosts");
+                            break;
+                        default:
+                            resourceGroupName = this.ResourceGroupName;
+                            hostGroupName = this.HostGroupName;
+                            hostName = this.Name;
+                            break;
+                    }
+
+                    var result = DedicatedHostsClient.DeleteWithHttpMessagesAsync(resourceGroupName, hostGroupName, hostName).GetAwaiter().GetResult();
+                    PSOperationStatusResponse output = new PSOperationStatusResponse
+                    {
+                        StartTime = this.StartTime,
+                        EndTime = DateTime.Now
+                    };
+
+                    if (result != null && result.Request != null && result.Request.RequestUri != null)
+                    {
+                        output.Name = GetOperationIdFromUrlString(result.Request.RequestUri.ToString());
+                    }
+
+                    WriteObject(output);
+                }
+            });
+        }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        public string HostGroupName { get; set; }
+
+        [Alias("HostName")]
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Position = 2,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true)]
+        [ResourceNameCompleter("Microsoft.Compute/dedicatedHosts", "ResourceGroupName")]
+        public string Name { get; set; }
+
+        [Parameter(
+            ParameterSetName = "DefaultParameter",
+            Mandatory = false)]
+        public SwitchParameter Force { get; set; }
```

Comment:
@hyonholee do we need the `-Force` parameter here? We only need to include this parameter if there is a reason to have an additional prompt for the user. If this command is deleting a resource _and_ its sub-resources, then I think it's OK to have the `-Force` since we want to prompt the user to confirm they want to delete the additional sub-resources.